File: src/main/java/io/gaegul/ch01/puzzle/Knight.java
package io.gaegul.ch01.puzzle;

import io.gaegul.ch01.puzzle.behavior.KnifeBehavior;

public class Knight extends Character {

	protected Knight() {
		super(new KnifeBehavior());
	}

	@Override
	public void fight() {
		System.out.println("나이트는 ");
		this.performWeapon();
		System.out.println("으로 전투합니다.");
	}
}

File: src/main/kotlin/io/gaegul/ch01/puzzle/Knight.kt
package io.gaegul.ch01.puzzle

import io.gaegul.ch01.puzzle.behavior.KnifeBehavior

class Knight: Character(
    KnifeBehavior()
) {
    override fun fight() {
        print("기사는 ")
        weaponBehavior.useWeapon()
        print("으로 전투할 수 있다.")
    }
}